test/model/src/driver/debug_log_parser.cpp

```diff
@@ -229,14 +258,74 @@ debug_log_parser::metadata_apply(const row_put &op)
 
     /* Special handling for the system prefix. */
     else if (starts_with(key, "system:")) {
-        /* We don't currently need to handle this. */
+
+        /* Set the base write generation. */
+        if (key == "system:checkpoint_base_write_gen") {
+            _base_write_gen = m->get_uint64("base_write_gen");
+        }
+
+        /* Handle checkpoints. */
+        else if (starts_with(key, "system:checkpoint") || starts_with(key, "system:oldest")) {
+            size_t p = key.find('.');
+            std::string ckpt_name = p == std::string::npos ? WT_CHECKPOINT : key.substr(p + 1);
+
+            /* Accumulate checkpoint metadata for future handling. */
+            auto &ckpt_metadata_map = _txn_ckpt_metadata[txn->id()];
+            auto i = ckpt_metadata_map.find(ckpt_name);
+            if (i == ckpt_metadata_map.end())
+                ckpt_metadata_map[ckpt_name] = m;
+            else
+                ckpt_metadata_map[ckpt_name] = config_map::merge(ckpt_metadata_map[ckpt_name], m);
+        }
+
+        /* Unsupported system URI. */
+        else
+            throw model_exception("Unsupported metadata system URI: " + key);
     }
 
     /* Otherwise this is an unsupported URI type. */
     else
         throw model_exception("Unsupported metadata URI: " + key);
 }
 
+/*
+ * debug_log_parser::metadata_checkpoint_apply --
+ *     Handle the given checkpoint metadata operation.
+ */
+void
+debug_log_parser::metadata_checkpoint_apply(
+  const std::string &name, std::shared_ptr<config_map> config)
+{
+    /* Get the stable timestamp. */
+    timestamp_t stable_timestamp = config->contains("checkpoint_timestamp") ?
+      config->get_uint64_hex("checkpoint_timestamp") :
+      k_timestamp_none;
+
+    /* Get the write generation number. */
+    write_gen_t write_gen = config->get_uint64("write_gen");
+
+    /* Get the transaction snapshot. */
+    kv_transaction_snapshot_ptr snapshot;
+    if (config->contains("snapshot_min")) {
+        txn_id_t snapshot_min = config->get_uint64("snapshot_min");
+        txn_id_t snapshot_max = config->get_uint64("snapshot_max");
+        std::shared_ptr<std::vector<uint64_t>> snapshot_ids;
+        if (config->contains("snapshots"))
+            snapshot_ids = config->get_array_uint64("snapshots");
+        else
+            snapshot_ids = std::make_shared<std::vector<uint64_t>>();
+        snapshot = std::make_shared<kv_transaction_snapshot_wt>(
+          write_gen, snapshot_min, snapshot_max, *snapshot_ids);
+    } else {
+        std::vector<uint64_t> snapshot_ids;
```

Comment:
Why is it declared as a simple vector here and as a shared pointer line 312?

Reply:
There is no need to make it a shared pointer, as it will be just deleted at the end of the function block. The constructor below does not keep the shared pointer but makes a copy of the value. Nonetheless, there is no need to declare an empty variable—I just moved the empty vector constructor inline with the snapshot constructor below.